src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+import {
+  getCommentsByPostId,
+  deleteComment,
+} from '../../api/comments';
+import { NewCommentForm } from '../NewCommentForm';
+import { Loader } from '../Loader';
+import { ErrorNotification } from '../ErrorNotification';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Error } from '../../types/Error';
+
+type Props = {
+  posts: Post[];
+  postId: number | null;
+  error: Error | null;
+  onError: (error: Error | null) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  posts,
+  postId,
+  error,
+  onError,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCommentForm, setIsCommentForm] = useState(false);
+
+  const handleAddComment = useCallback((newComment) => {
+    setComments(allComments => [...allComments, newComment]);
+  }, []);
+
+  const currentPost
+    = useMemo(() => posts.filter(post => post.id === postId)[0], [postId]);
+
+  const handleDeleteComment = async (commentId: number) => {
+    try {
+      setComments(
+        allComments => allComments.filter(
+          comment => comment.id !== commentId,
+        ),
+      );
+      onError(null);
+
+      if (!commentId) {
```

Comment:
Empty return statement in handleDeleteComment function might cause issues. Consider throwing an error or returning a default value when commentId is not provided.